src/App.tsx

```diff
@@ -1,24 +1,220 @@
+    = todoList?.filter(todo => todo.completed).map(todo => todo.id);
+
+    if (completedTodo) {
+      setLoadingTodoIds(completedTodo);
+    }
+
+    completedTodo?.forEach(todo => {
+      deleteTodo(todo)
+        .then(() => {
+          if (todoList) {
+            setTodoList(
+              todoList.filter(todoFilter => {
+                return !completedTodo.includes(todoFilter.id);
+              }),
+            );
+          }
+        })
+        .catch(() => {
+          setTypeError(Errors.REMOVE);
+        });
+    });
+  };
+
+  const patchHandlerTodoCompleted = (
+    id: number,
+    completed: boolean,
+    bringInList: boolean,
+  ) => {
+    patchTodoCompleted(id, completed)
+      .then(() => {
+        setLoaderTodo(0);
+
+        if (bringInList) {
+          const newArray = todoList?.map(todo => {
+            const newTodo = { ...todo };
+
+            if (todo.id === id) {
+              newTodo.completed = completed;
+
+              return newTodo;
+            }
+
+            return newTodo;
+          });
+
+          if (newArray) {
+            setTodoList(newArray);
+          }
+        }
+      })
+      .catch(() => {
+        setTypeError(Errors.UPDATE);
+      });
+  };
+
+  const patchHandlerTodoTitle = (
+    id: number,
+    title: string,
+  ) => {
+    patchTodoTitle(id, title)
+      .then(() => {
+        setLoaderTodo(0);
+
+        const newArray = todoList?.map(todo => {
+          const newTodo = { ...todo };
+
+          if (todo.id === id) {
+            newTodo.title = title;
+          }
+
+          return newTodo;
+        });
+
+        if (newArray) {
+          setTodoList(newArray);
+        }
+      })
+      .catch(() => {
+        setTypeError(Errors.UPDATE);
+      });
+  };
+
+  const toggleAllHandler = () => {
+    const newArray = todoList.map(todo => {
+      const newTodo = { ...todo };
+
+      if (newTodo.completed === completedAll) {
+        patchHandlerTodoCompleted(newTodo.id, !completedAll, false);
+        newTodo.completed = !completedAll;
+      }
+
+      return newTodo;
+    });
+
+    setCompletedAll(!completedAll);
+
+    if (newArray) {
+      setTodoList(newArray);
+    }
+  };
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(setTodoList)
+      .catch(() => {
+        setTypeError(Errors.UPDATE);
+      });
+  }, []);
+
   if (!USER_ID) {
     return <UserWarning />;
   }
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+
+      <div className="todoapp__content">
+        <Header
+          USER_ID={USER_ID}
```

Comment:
```suggestion
          userId={USER_ID}
```
It's better to do it in this way :arrow_up: